meteor-slingshot/lib/upload.js

```diff
@@ -202,13 +202,22 @@ Slingshot.Upload = function (directive, metaData) {
         );
       });
 
-      xhr.open('POST', self.instructions.upload, true);
+      let isR2 = self.instructions.upload.includes('r2.cloudflarestorage.com');
+      let method = 'POST';
+      let formData = buildFormData();
+
+      if (isR2) {
+        method = 'PUT';
+        formData = self.file;
```

Comment:
Why not use the `formData` from `buildFormData`?

Reply:
R2 has the peculiarity of not using multi part form data. That's why we don't need to call buildFormData for it (which will append the file to a FormData).
As Cloudflare doesn't support POST requests directly, we need to make a single PUT call to the signed url containing the file that we want to upload for r2.

Check these docs:

https://developers.cloudflare.com/r2/api/s3/presigned-urls/#supported-http-methods